### ISSUE TITLE
Project Config Apply command does not work

### ISSUE THREAD

**Issue by TheFox** [craftcms/cms#7481]
### Description

The `./craft project-config/apply` command does not work properly. We have a few diffs in the Project Config, and when we want to apply them, the command does not throw any error or so, but after the Apply command, the diff is still the same on the command line, but on the web CMS <https://domain/admin/utilities/project-config> we do not see any diffs.

See this command line:

```bash
production:/app# ./craft project-config/diff
      structure:
        maxLevels: null
        uid: cb4c9cbe-46f5-4362-9683-d52e4246ada6
- dateModified: 1611838098
+ dateModified: 1612264284
  dolphiq_sitemap_entries:
    201841a3-809b-43c5-be30-b94f33046a76:
      changefreq: hourly
...
      contentColumnType: string
      fieldGroup: 252831be-a183-4ddb-a03e-dce04e3d8ccf
      handle: abstandStil
-     instructions: ''
+     instructions: 'Fügt einen Abstand hinzu'
      name: 'Abstand Stil'
      searchable: true
      settings:

production:/app# ./craft project-config/apply
Applying changes from your project config files ... done

production:/app# ./craft project-config/diff
      structure:
        maxLevels: null
        uid: cb4c9cbe-46f5-4362-9683-d52e4246ada6
- dateModified: 1611838098
+ dateModified: 1612264284
  dolphiq_sitemap_entries:
    201841a3-809b-43c5-be30-b94f33046a76:
      changefreq: hourly
...
      contentColumnType: string
      fieldGroup: 252831be-a183-4ddb-a03e-dce04e3d8ccf
      handle: abstandStil
-     instructions: ''
+     instructions: 'Fügt einen Abstand hinzu'
      name: 'Abstand Stil'
      searchable: true
      settings:
```

### Additional info

- Craft version: 3.5.16
- PHP version: 7.2.34
- Database driver & version: MySQL 5.5

**Comment by brandonkelly** [comment]
Hm strange. I wonder if this was just a latency issue? The diff view in the Project Config utility uses the same underlying diffing function as the terminal command. After checking the utility, did you check the `project-config/diff` command again?

**Comment by TheFox** [comment]
Yes, we did. As you can see in my shell snippet, we run the diff again.

1. We run `./craft project-config/diff` — says we have.
2. We run `./craft project-config/apply` to apply the diffs.
3. We run `./craft project-config/diff` — still says we have diffs.

**Comment by brandonkelly** [comment]
Did the new field instructions take effect?

**Comment by TheFox** [comment]
Yes, they took effect.

**Comment by dr0bz** [comment]
Hi @brandonkelly,

is it a bug? We don't have this behavior on other instances within our eco system. After the changes are applied the diff becomes empty. This is the behavior i would expect.

One difference on this instance to the others, which goes to mine mind, is that the `allowAdminChanges` are disabled. 

Best regards

**Comment by brandonkelly** [comment]
Is there any chance we can get SSH access to this environment? I’m stumped as to why the CLI command would still be showing an outdated diff. If so please send access credentials to support@craftcms.com.
